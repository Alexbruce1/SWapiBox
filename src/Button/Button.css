.button {
  background-color: #E5E5EB;
  border: 4px solid #5468B1;
  border-radius: 10px;
  color: #5468B1;
  cursor: pointer;
  display: inline-flex;
  height: 2rem;
  padding: 8px 1rem;
  transition-duration: .5s;
  vertical-align: middle;
}

.button:hover {
  /* box-shadow: 0 .5px 1px #000; */
  border: 4px solid white;
  box-shadow: inset 0 0 20px #1CA512, 0 0 20px #1CA512;
}

/* .button:hover .favorite-button-icon {
  filter: invert(100%);
} */

.button:hover .button-label {
  /* color: white; */
}

.favorite-button-icon {
  height: 2rem;
}

.button-label {
  margin: 0 5px;
}