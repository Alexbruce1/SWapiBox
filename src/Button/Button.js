import React from 'react';
import './Button.css';
import PropTypes from 'prop-types';

const Button = ({ purpose, image, setButtonName, count }) => {

  const handleClick = (event) => {
    let name = event.target.getAttribute('name');
    setButtonName(name);
  }
  
  const favoritesCount = () => {
    console.log('COUNT: ', count.length)
    if (count.length) {
      return <h2>({count.length})</h2>;
    } else {
      return <h2>(0)</h2>;
    }
  }

  return (
    <div className="button" onClick={handleClick} name={purpose} >
      <img name={purpose} className="favorite-button-icon" alt="img" src={image} />
      <h2 name={purpose} className="button-label" >
        {purpose}
      </h2>
        { favoritesCount }
    </div>
  )
}


Button.propTypes = {
  purpose: PropTypes.string.isRequired,
  count: PropTypes.array,
  // image: PropTypes.
}

export default Button;