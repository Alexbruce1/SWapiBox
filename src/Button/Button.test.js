import React from 'react';
import { configure, shallow } from 'enzyme';


import Adapter from 'enzyme-adapter-react-16';
import Button from './Button';

configure({ adapter: new Adapter() });



describe('Button', () => {
  let mockPurpose = "people"
  let mockImage;
  let mockFunc = jest.fn();
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Button purpose={mockPurpose} />)
  })

  it.skip('Should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})