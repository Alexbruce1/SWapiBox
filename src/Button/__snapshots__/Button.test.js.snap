// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    purpose="people"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="img"
          className="favorite-button-icon"
          name="people"
        />,
        <h2
          className="button-label"
          name="people"
        >
          people
        </h2>,
        [Function],
      ],
      "className": "button",
      "name": "people",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "img",
          "className": "favorite-button-icon",
          "name": "people",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "people",
          "className": "button-label",
          "name": "people",
        },
        "ref": null,
        "rendered": "people",
        "type": "h2",
      },
      [Function],
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="img"
            className="favorite-button-icon"
            name="people"
          />,
          <h2
            className="button-label"
            name="people"
          >
            people
          </h2>,
          [Function],
        ],
        "className": "button",
        "name": "people",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "img",
            "className": "favorite-button-icon",
            "name": "people",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "people",
            "className": "button-label",
            "name": "people",
          },
          "ref": null,
          "rendered": "people",
          "type": "h2",
        },
        [Function],
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
